src/unicorn/builder.rs

```diff
@@ -531,6 +531,18 @@ impl ModelBuilder {
         self.reg_flow_ite(itype.rd(), sra_node);
     }
 
+    fn model_sllw(&mut self, rtype: RType) {
+        let thirtytwo = self.new_const(32);
+
+        // shift value should only be a value less or equal than 2^32
+        let mut shifted_r2 = self.new_sll(self.reg_node(rtype.rs2()), thirtytwo.clone());
```

Comment:
The RISC-V spec says ...
```
SLLW, SRLW, and SRAW are RV64I-only instructions that are analogously defined but operate
on 32-bit values and produce signed 32-bit results. The shift amount is given by rs2[4:0]
```
The way I interpret this we need to mask out more from `rs2` and only use that lowest five bits in `rs2` as a shift amount. What do you think?

Reply:
You can look at how I implemented `sll` which does something similar and only uses the lowest six bits ...
```
    fn model_sll(&mut self, rtype: RType) {
        // Only the low 6 bits of rs2 are considered for the shift amount.
        let mask_node = self.new_const(0x3f); // TODO: Make this a global constant.
        let amount_node = self.new_and_word(self.reg_node(rtype.rs2()), mask_node);
        let sll_node = self.new_sll(self.reg_node(rtype.rs1()), amount_node);
        self.reg_flow_ite(rtype.rd(), sll_node);
    }
```

Reply:
thanks! working on it!